VTUL/dlp-access
# start fetching subjects

**Show existing values for subjects and let admin select them for site, while also allowing admin to add new values.**
* * *

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2471) (:star:)

* Other Relevant Links (Meeting note, project page, related pull requests, etc.)

# What does this Pull Request do? (:star:)
Show existing values for subjects and let admin select them for site, while also allowing admin to add new values.

# What's the changes? (:star:)
* Fetch ALL subjects and use that to build a list for admins to select from
* Add input boxes to allow adding additional subject
* Remove any duplicate subjects before submitting

# How should this be tested?
* Visit "/siteAdmin" page
* Click "Search Page Config" and then click "Edit" radio button.
* Scroll down to almost the bottom of the page and select "Subject" from the dropdown labeled "Add New Search Facet"
* click "Add New Search Facet"
* Click a few checkboxes to select some subjects
* Scroll to the bottom and click "Update Facet and Sort Fields"
* Check that the subject(s) you clicked are added to your sites Search facets

# Additional Notes:
* branch: `LIBTD-2471`

# Interested parties
@yinlinchen 

(:star:) Required fields
